scvi/dataset/synthetic.py

```diff
@@ -42,3 +42,113 @@ def __init__(self, mu=4.0, theta=2.0, dropout=0.7, save_path='data/'):
+        :param n_labels: (useless for now) Number of labels
+        :param mode: Should be either
+            - 'nb': mode without zero inflation
+            - 'zi': mode including zero inflation for ALL genes
+            -'mixed': mode in which SOME genes are Zero-Inflated while some other are not
+            The ratio_genes_zi parameter allow to control the proportion of genes that are going
+            to be ZI
+        :param ratio_genes_zi: Proportion of ZI genes
+        """
+
+        assert mode in ['mixed', 'zi', 'nb'], 'Mode {} not recognized'.format(mode)
+        np.random.seed(0)
+
+        if n_genes_total % n_clusters > 0:
+            print("Warning, clusters have inequal sizes")
+
+        if n_genes_high > (n_genes_total // n_clusters):
+            print("Overlap of", n_genes_high - (n_genes_total // n_clusters), "genes")
+
+        # Generate data before dropout
+        batch_size = n_cells_cluster * n_clusters
+        data = np.ones((n_batches, batch_size, n_genes_total))
+
+        # For each cell cluster, some genes have a high expression, the rest
+        # has a low expression. The scope of high expression genes "moves"
+        # with the cluster
+        for cluster in range(n_clusters):
+            ind_first_gene_cluster = cluster * (n_genes_total // n_clusters)
+            ind_last_high_gene_cluster = ind_first_gene_cluster + n_genes_high
+
+            # Weights in a cluster to create highly-expressed and low-expressed genes
+            weights = weight_low * np.ones((n_genes_total,))
+            weights[ind_first_gene_cluster:ind_last_high_gene_cluster] = weight_high
+            weights /= weights.sum()
+
+            exprs = np.random.poisson(lam_0 * weights, size=(n_batches, n_cells_cluster,
+                                                             n_genes_total))
+            data[:, cluster * n_cells_cluster:(cluster + 1) * n_cells_cluster, :] = exprs
+        # Apply dropout depending on the mode
+        if mode == "zi":
+            mask = np.random.binomial(n=1, p=1 - p_dropout,
+                                      size=(n_batches, batch_size, n_genes_total))
+            newdata = (data * mask)  # We put the batch index first
+        elif mode == 'nb':
+            newdata = data
+        else:
+            assert ratio_genes_zi is not None
+            assert ratio_genes_zi <= 1.
+            n_genes_zi = int(n_genes_total * ratio_genes_zi)
+
+            submask = np.random.binomial(n=1, p=1 - p_dropout,
+                                         size=(n_batches, batch_size, n_genes_zi))
+            mask = np.ones_like(data)
+            random_permutation = np.random.permutation(np.arange(n_genes_total))
+            self.zi_genes_idx = random_permutation[:n_genes_zi]
+            mask[:, :, random_permutation[:n_genes_zi]] = submask
+
+            newdata = data * mask
+        labels = np.random.randint(0, n_labels, size=(n_batches, batch_size, 1))
```

Comment:
How about giving all the cells in each cluster the same label? Then we can plot the tsne of our latent space to confirm that all the cells with the same label are clustered together. (The labels are used to color code the tnse plots in one of our functions)

Reply:
Great, thanks for this advice!